src/main/java/info/batey/kafka/unit/KafkaUnit.java

```diff
@@ -98,32 +149,63 @@ private static int parseConnectionString(String connectionString) {
         }
     }
 
-    private static int getEphemeralPort() throws IOException {
+    private static int getEphemeralPort() {
         try (ServerSocket socket = new ServerSocket(0)) {
             return socket.getLocalPort();
+        } catch (IOException e) {
+            throw new RuntimeException("Failed finding an available port: "+e.getMessage(), e);
         }
     }
 
     public void startup() {
-        zookeeper = new Zookeeper(zkPort);
-        zookeeper.startup();
+        startZookeeper();
+        zkClient = new ZkClient(getZookeeperString(), 30000, 30000, ZKStringSerializer$.MODULE$);
```

Comment:
Good comment. This took me literally hours to debug why on earth I can't send messages to the brokers. Without it the zkClient simply doesn't write the correct data to ZK. Once you use it, it actually works. I've added the following comment for future developers:
```
        // ZKStringSerializer$.MODULE$ - this obscure object creation is what makes this client work
        // Without it , the topic is reported as created but is actually not usable and produces very hard
        // to understand behaviour
```